### ISSUE TITLE
morphology.skeletonize junction handling

### ISSUE THREAD

**Issue by erberj** [scikit-image/scikit-image#6828]
### Description:

On a black and white image, if i have 2 crossing fibre like contours (long, steady and thin), the funktion above creates junctions like

1)    ">---<"   instead of 
2)    "><"

chancing a junction with 4 ways to 2 junctions with 3 ways.

Could we add a method do join 1) into 2) in the skeleton? This would make detecting single fibres in a cluster of fibres more easy.

**Comment by jakeMartin1234** [comment]
Looks as though all of the logic behind skeletonization is in this code segment. If you look just below it in the loop there is logic behind how it is used to decide which pixels get to stay white.
https://github.com/scikit-image/scikit-image/blob/907bfb71d2c8e4d58c2dffc6056760c1e00d070b/skimage/morphology/_skeletonize_cy.pyx#L43-L61

**Comment by jni** [comment]
@erberj @jakeMartin1234 messing with the core skeletonization algorithm is generally hard. There are other skeletonization methods that we haven't implemented in scikit-image yet, see e.g. TEASAR #1621. You might want to see if you can get those working with your data @erberj! We would definitely welcome a contribution adding that method to scikit-image!

**Comment by jni** [comment]
@erberj I agree with @jakeMartin1234 — it is very useful for us to be able to help you if you send us the raw data and code you used to generate your results.

**Comment by erberj** [comment]
The Code is heavily embedded in a greater Progamm and hard to extract, but basicly this is what happens:
- get raw picture (glas fibres)
- cv.threshhold it to black and white (127,255)
- remove small objects
- create skeleton
- show skeleteton with cv.imshow

I think creating ">---<" like junctions is correct in cases of long overlapping junctions, but not for a straight junction with a angle of 30-90 degrees

**Comment by jni** [comment]
> not for a straight junction with a angle of 30-90 degrees

Well, how do you pick 30 as the limit and not 29.5 or 35.7? ;) Once the angle becomes too shallow to represent with four pixels, (ie very quickly away from 90º), the algorithm is going to prefer to get rid of pixels rather than try to maintain a long line.

btw have you tried the medial axis transform to see if it performs better? And TEASAR?

> but basicly this is what happens:

If you can reproduce the above images with the *raw picture* and those 5 function calls, and share the raw picture with us, then we will be in a good starting point. Currently, we don't even have access to a good picture with which to produce such junctions. (A screenshot is not as valuable because it is lossy compared to the original dataset due to aliasing.)

**Comment by erberj** [comment]
This is the raw image:
bmp or tif are not supported here

![LID_107416_1_AOI](https://user-images.githubusercontent.com/40894308/227727539-3b165037-f05f-4dc5-ba2f-f39ece9a9efa.jpg)
